org.cryptomator.Cryptomator.yaml

```diff
@@ -0,0 +1,100 @@
+app-id: org.cryptomator.Cryptomator
+command: cryptomator
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk17
+separate-locales: false
+finish-args:
+  - --device=dri
+  - --env=PATH=/app/bin/:/app/jre/bin/:/usr/bin/
+  - --filesystem=home
+  - --share=ipc
+  - --share=network
+  - --socket=session-bus
```

Comment:
you should declare the require talk name instead. allowing the session-bus is kinda no no

Reply:
@purejava you probably know what is needed here?

Reply:
The required talk name `--talk-name=org.freedesktop.secrets` is in place.
This should work for the `org.freedesktop.Platform` in conjunction with `- shared-modules/libsecret/libsecret.json` (which is missing here) or with `org.gnome.Platform`, but I tried both and neither does without the `--socket=session-bus` permission.

I logged the apps access to d-bus, but could not see what is missing in the talk-names:
```
ralph@fusion flathub % flatpak run --log-session-bus org.cryptomator.Cryptomator                                                                                 
18:46:49.477 [main] DEBUG org.cryptomator.common.Environment - user.home: /home/ralph
18:46:49.483 [main] DEBUG org.cryptomator.common.Environment - java.library.path: :/app/Cryptomator/lib/app:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
18:46:49.483 [main] DEBUG org.cryptomator.common.Environment - user.language: de
18:46:49.483 [main] DEBUG org.cryptomator.common.Environment - user.region: null
18:46:49.483 [main] DEBUG org.cryptomator.common.Environment - logback.configurationFile: null
18:46:49.483 [main] DEBUG org.cryptomator.common.Environment - cryptomator.settingsPath: ~/.config/Cryptomator/settings.json:~/.Cryptomator/settings.json
18:46:49.483 [main] DEBUG org.cryptomator.common.Environment - cryptomator.ipcSocketPath: ~/.config/Cryptomator/ipc.socket
18:46:49.483 [main] DEBUG org.cryptomator.common.Environment - cryptomator.keychainPath: null
18:46:49.483 [main] DEBUG org.cryptomator.common.Environment - cryptomator.logDir: ~/.local/share/Cryptomator/logs
18:46:49.483 [main] DEBUG org.cryptomator.common.Environment - cryptomator.pluginDir: null
18:46:49.483 [main] DEBUG org.cryptomator.common.Environment - cryptomator.mountPointsDir: ~/.local/share/Cryptomator/mnt
18:46:49.483 [main] DEBUG org.cryptomator.common.Environment - cryptomator.minPwLength: null
18:46:49.483 [main] DEBUG org.cryptomator.common.Environment - cryptomator.appVersion: 1.6.5
18:46:49.483 [main] DEBUG org.cryptomator.common.Environment - cryptomator.buildNumber: flatpak-1
18:46:49.483 [main] DEBUG org.cryptomator.common.Environment - cryptomator.showTrayIcon: false
18:46:49.483 [main] DEBUG org.cryptomator.common.Environment - fuse.experimental: false
18:46:49.524 [main] DEBUG org.cryptomator.common.ShutdownHook - Registered shutdown hook.
18:46:49.643 [main] DEBUG org.cryptomator.common.settings.SettingsProvider - Attempting to load settings from /home/ralph/.config/Cryptomator/settings.json
18:46:49.678 [main] INFO org.cryptomator.common.settings.SettingsProvider - Settings loaded from /home/ralph/.config/Cryptomator/settings.json
18:46:49.680 [main] INFO  org.cryptomator.launcher.Cryptomator - Starting Cryptomator 1.6.5 on Linux 5.16.9-arch1-1 (amd64)
18:46:49.712 [main] INFO  org.cryptomator.ipc.Server - Spawning IPC server listening on socket /home/ralph/.config/Cryptomator/ipc.socket
Gtk-Message: 18:46:49.880: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:46:49.880: Failed to load module "canberra-gtk-module"
18:46:50.259 [JavaFX Application Thread] INFO  o.c.ui.launcher.FxApplicationStarter - JavaFX Runtime started.
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake AddMatch for org.freedesktop.Notifications
C3: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.Notifications
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for org.cryptomator.Cryptomator
C5: -> org.freedesktop.DBus fake wildcarded AddMatch for org.gtk.vfs
C6: -> org.freedesktop.DBus fake AddMatch for org.freedesktop.secrets
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.secrets
C8: -> org.freedesktop.DBus fake AddMatch for org.freedesktop.Flatpak
C9: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.Flatpak
C10: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.org.cryptomator.Cryptomator
C11: -> org.freedesktop.DBus fake AddMatch for org.freedesktop.DBus.ListActivatableNames
C12: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus.ListActivatableNames
C13: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C14: -> org.freedesktop.DBus fake ListNames
B1: <- org.freedesktop.DBus return from C1
B2: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B3: <- org.freedesktop.DBus return from C2
*SKIPPED*
B4: <- org.freedesktop.DBus return from C3
*SKIPPED*
B5: <- org.freedesktop.DBus return from C4
*SKIPPED*
B6: <- org.freedesktop.DBus return from C5
*SKIPPED*
B7: <- org.freedesktop.DBus return from C6
*SKIPPED*
B8: <- org.freedesktop.DBus return from C7
*SKIPPED*
B9: <- org.freedesktop.DBus return from C8
*SKIPPED*
B10: <- org.freedesktop.DBus return from C9
*SKIPPED*
B11: <- org.freedesktop.DBus return from C10
*SKIPPED*
B12: <- org.freedesktop.DBus return from C11
*SKIPPED*
B13: <- org.freedesktop.DBus return error org.freedesktop.DBus.Error.NameHasNoOwner from C12
*SKIPPED*
B14: <- org.freedesktop.DBus return from C13
*SKIPPED*
B15: <- org.freedesktop.DBus return from C14
C15: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C16: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.Notifications
C17: -> org.freedesktop.DBus fake GetNameOwner for org.gtk.vfs.Daemon
C18: -> org.freedesktop.DBus fake GetNameOwner for org.gtk.vfs.UDisks2VolumeMonitor
C19: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.Flatpak
C20: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C21: -> org.freedesktop.DBus fake GetNameOwner for org.gtk.vfs.GPhoto2VolumeMonitor
C22: -> org.freedesktop.DBus fake GetNameOwner for org.gtk.vfs.AfcVolumeMonitor
C23: -> org.freedesktop.DBus fake GetNameOwner for org.gtk.vfs.mountpoint_1239
C24: -> org.freedesktop.DBus fake GetNameOwner for org.gtk.vfs.mountpoint_1416
C25: -> org.freedesktop.DBus fake GetNameOwner for org.gtk.vfs.GoaVolumeMonitor
C26: -> org.freedesktop.DBus fake GetNameOwner for org.gtk.vfs.Metadata
C27: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.secrets
C28: -> org.freedesktop.DBus fake GetNameOwner for org.gtk.vfs.MTPVolumeMonitor
*SKIPPED*
B16: <- org.freedesktop.DBus return from C15
*SKIPPED*
B17: <- org.freedesktop.DBus return from C16
*SKIPPED*
B18: <- org.freedesktop.DBus return from C17
*SKIPPED*
B19: <- org.freedesktop.DBus return from C18
*SKIPPED*
B20: <- org.freedesktop.DBus return from C19
*SKIPPED*
B21: <- org.freedesktop.DBus return from C20
*SKIPPED*
B22: <- org.freedesktop.DBus return from C21
*SKIPPED*
B23: <- org.freedesktop.DBus return from C22
*SKIPPED*
B24: <- org.freedesktop.DBus return from C23
*SKIPPED*
B25: <- org.freedesktop.DBus return from C24
*SKIPPED*
B26: <- org.freedesktop.DBus return from C25
*SKIPPED*
B27: <- org.freedesktop.DBus return from C26
*SKIPPED*
B28: <- org.freedesktop.DBus return from C27
*SKIPPED*
B29: <- org.freedesktop.DBus return from C28
*SKIPPED*
C29: -> org.freedesktop.DBus call org.freedesktop.DBus.ListActivatableNames at /org/freedesktop/DBus
B30: <- org.freedesktop.DBus return from C29
C4: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C5: -> org.freedesktop.DBus fake AddMatch for org.freedesktop.Notifications
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.Notifications
C7: -> org.freedesktop.DBus fake wildcarded AddMatch for org.cryptomator.Cryptomator
C8: -> org.freedesktop.DBus fake wildcarded AddMatch for org.gtk.vfs
C9: -> org.freedesktop.DBus fake AddMatch for org.freedesktop.secrets
C10: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.secrets
C11: -> org.freedesktop.DBus fake AddMatch for org.freedesktop.Flatpak
C12: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.Flatpak
C13: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.org.cryptomator.Cryptomator
C14: -> org.freedesktop.DBus fake AddMatch for org.freedesktop.DBus.ListActivatableNames
C15: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus.ListActivatableNames
C16: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C17: -> org.freedesktop.DBus fake ListNames
B1: <- org.freedesktop.DBus return from C4
B2: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B3: <- org.freedesktop.DBus return from C5
*SKIPPED*
B4: <- org.freedesktop.DBus return from C6
*SKIPPED*
B5: <- org.freedesktop.DBus return from C7
*SKIPPED*
B6: <- org.freedesktop.DBus return from C8
*SKIPPED*
B7: <- org.freedesktop.DBus return from C9
*SKIPPED*
B8: <- org.freedesktop.DBus return from C10
*SKIPPED*
B9: <- org.freedesktop.DBus return from C11
*SKIPPED*
B10: <- org.freedesktop.DBus return from C12
*SKIPPED*
B11: <- org.freedesktop.DBus return from C13
*SKIPPED*
B12: <- org.freedesktop.DBus return from C14
*SKIPPED*
B13: <- org.freedesktop.DBus return error org.freedesktop.DBus.Error.NameHasNoOwner from C15
*SKIPPED*
B14: <- org.freedesktop.DBus return from C16
*SKIPPED*
B15: <- org.freedesktop.DBus return from C17
C18: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C19: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.Notifications
C20: -> org.freedesktop.DBus fake GetNameOwner for org.gtk.vfs.Daemon
C21: -> org.freedesktop.DBus fake GetNameOwner for org.gtk.vfs.UDisks2VolumeMonitor
C22: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.Flatpak
C23: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C24: -> org.freedesktop.DBus fake GetNameOwner for org.gtk.vfs.GPhoto2VolumeMonitor
C25: -> org.freedesktop.DBus fake GetNameOwner for org.gtk.vfs.AfcVolumeMonitor
C26: -> org.freedesktop.DBus fake GetNameOwner for org.gtk.vfs.mountpoint_1239
C27: -> org.freedesktop.DBus fake GetNameOwner for org.gtk.vfs.mountpoint_1416
C28: -> org.freedesktop.DBus fake GetNameOwner for org.gtk.vfs.GoaVolumeMonitor
C29: -> org.freedesktop.DBus fake GetNameOwner for org.gtk.vfs.Metadata
C30: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.secrets
C31: -> org.freedesktop.DBus fake GetNameOwner for org.gtk.vfs.MTPVolumeMonitor
*SKIPPED*
B16: <- org.freedesktop.DBus return from C18
*SKIPPED*
B17: <- org.freedesktop.DBus return from C19
*SKIPPED*
B18: <- org.freedesktop.DBus return from C20
*SKIPPED*
B19: <- org.freedesktop.DBus return from C21
*SKIPPED*
B20: <- org.freedesktop.DBus return from C22
*SKIPPED*
B21: <- org.freedesktop.DBus return from C23
*SKIPPED*
B22: <- org.freedesktop.DBus return from C24
*SKIPPED*
B23: <- org.freedesktop.DBus return from C25
*SKIPPED*
B24: <- org.freedesktop.DBus return from C26
*SKIPPED*
B25: <- org.freedesktop.DBus return from C27
*SKIPPED*
B26: <- org.freedesktop.DBus return from C28
*SKIPPED*
B27: <- org.freedesktop.DBus return from C29
*SKIPPED*
B28: <- org.freedesktop.DBus return from C30
*SKIPPED*
B29: <- org.freedesktop.DBus return from C31
*SKIPPED*
C32: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B30: <- org.freedesktop.DBus return from C32
C34: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B31: <- org.freedesktop.DBus return from C34
C35: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B32: <- org.freedesktop.DBus return from C35
C36: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B33: <- org.freedesktop.DBus return from C36
C37: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B34: <- org.freedesktop.DBus return from C37
C38: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B35: <- org.freedesktop.DBus return from C38
C39: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B36: <- org.freedesktop.DBus return from C39
C40: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B37: <- org.freedesktop.DBus return from C40
C41: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B38: <- org.freedesktop.DBus return from C41
C42: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B39: <- org.freedesktop.DBus return from C42
C43: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B40: <- org.freedesktop.DBus return from C43
C44: -> org.freedesktop.DBus call org.freedesktop.DBus.ListActivatableNames at /org/freedesktop/DBus
B41: <- org.freedesktop.DBus return from C44
```

Reply:
Ok, I did some more investigation on this.
Running the app on an KDE environment with these sole changes does allow to use d-bus, access and use kwallet:
```
ralph@ubuntu:~/git/flathub$ git diff
diff --git a/org.cryptomator.Cryptomator.yaml b/org.cryptomator.Cryptomator.yaml
index 1995673..22055f8 100644
--- a/org.cryptomator.Cryptomator.yaml
+++ b/org.cryptomator.Cryptomator.yaml
@@ -12,11 +12,11 @@ finish-args:
   - --filesystem=home
   - --share=ipc
   - --share=network
-  - --socket=session-bus
   - --socket=x11
   - --talk-name=org.freedesktop.Flatpak
   - --talk-name=org.freedesktop.Notifications
   - --talk-name=org.freedesktop.secrets
+  - --talk-name=org.kde.kwalletd5
   - --talk-name=org.gtk.vfs.*
   - --talk-name=org.gtk.vfs
 cleanup:
ralph@ubuntu:~/git/flathub$
```

The app implementations for secret-service and kdewallet are very similar. The only real difference that comes to my mind is that secret-service communicates encrypted with d-bus. Might this cause this issue?